config/eurotronic/eur_spiritz.xml

```diff
@@ -1,83 +1,110 @@
+Product website: https://www.eurotronic.org/produkte/spirit-z-wave-plus.html
+Tech manual: https://www.eurotronic.org/fileadmin/user_upload/eurotronic.org/Produktbilder/spirit_z_wave_plus/Spirit_Z-Wave_BAL_web_EN_view_04.pdf
+-->
 <Product xmlns='http://code.google.com/p/open-zwave/'>
-    <CommandClass id="67" base="0" typeInterpretation="A" />
-    <CommandClass id="64" >
+    <!-- Multilevel switch for setting valve opening -->
+    <CommandClass id="38">
+        <Value type="byte" genre="user" index="0" label="Valve Opening" units="%" min="0" max="100" value="0" />
+    </CommandClass>
+
+    <!-- Thermostat mode -->
+    <CommandClass id="64">
         <Instance index="1" />
         <Value type="list" genre="user" instance="1" index="0" label="Mode">
-                <Item label="Off" value="0" />
-                <Item label="Heat" value="1" />
-                <Item label="Heat Econ" value="11" />
-                <Item label="Full Power" value="15" />
-                <Item label="Manufacturer Specific" value="31" />
+            <Help>
+                Off: No heating, only frost protection.
+                Auto: Room temperature will be kept at the configured setpoint.
+                Eco: Energy save heating mode. Room temperature will be lowered to the configured eco setpoint in order to save energy.
+                Full Power: Full power heating. This mode is left automatically after 5 minutes.
+                Manufacturer Specific: Direct valve control mode. The valve opening percentage can be controlled using the switch multilevel command class.
+            </Help>
+            <Item label="Off" value="0" />
+            <Item label="Auto" value="1" />
+            <Item label="Eco" value="11" />
+            <Item label="Full Power" value="15" />
+            <Item label="Manufacturer Specific" value="31" />
         </Value>
         <SupportedModes>
-                <Mode index="0" label="Off" />
-                <Mode index="1" label="Heat" />
-                <Mode index="11" label="Heat Econ" />
-                <Mode index="15" label="Full Power" />
-                <Mode index="31" label="Manufacturer Specific" />
+            <Mode index="0" label="Off" />
+            <Mode index="1" label="Auto" />
+            <Mode index="11" label="Eco" />
+            <Mode index="15" label="Full Power" />
+            <Mode index="31" label="Manufacturer Specific" />
         </SupportedModes>
     </CommandClass>
 
+    <!-- Setpoints -->
+    <CommandClass id="67" base="0" typeInterpretation="A">
+        <Instance index="1"/>
+        <Value type="decimal" genre="user" instance="1" index="1" label="Target Temperature" units="&#x00b0;C" read_only="false" write_only="false" min="8" max="28" value="20"/>
+        <Value type="decimal" genre="user" instance="1" index="11" label="Target Temperature Eco" units="&#x00b0;C" read_only="false" write_only="false" min="8" max="28" value="16"/>
```

Comment:
What up with that encoding of the degree degree sign?
Got curious and looked through some other config files. They were all just "C" instead of "°C". I Would personally prefer the degree sing, but I think it would still be good to also do it that way.

Reply:
Putting the Unicode `°` sign will produce invalid XML, hence it has to be escaped. Using `°C` instead of just `C` is more specific and also looks really good in UIs. My bet is, that until now just nobody cared enough about it to look up the escape sequence for this Unicode character. I'm passionate about improving things, so I'd vote for leaving it in there, as it might inspire others to adopt it as well :) But it's your call.